### ISSUE TITLE
[TopAppBar] Title not marked as heading for accessibility

### ISSUE THREAD

**Issue by albertvaka** [material-components/material-components-android#4073]
**Expected behavior:** The title `TextView` of the toolbar should be marked as a heading so screen readers describe it as so.

**Source code:** 
* https://github.com/material-components/material-components-android/blob/master/docs/components/TopAppBar.md
* https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/appbar/MaterialToolbar.java

**Android API version:** Any

**Material Library version:** Any

**Device:** Any

**Comment by albertvaka** [comment]
Note: Apparently there's a different way from headings [1] to designate titles [2], which we should use instead.

[1] https://www.w3.org/WAI/WCAG21/Understanding/section-headings.html
[2] https://www.w3.org/WAI/WCAG21/Understanding/page-titled
